### PR TITLE
Fixes #508 Adds singularity runner

### LINKED ISSUE #508: cli: add singularity runner
Implement a `SingularityRunner` class that runs actions on singularity containers

### PR DESCRIPTION
**Pull request by JayjeetAtGithub**
Adds a `SingularityRunner` class in `gha.py` to run singularity based action.

### CODE REVIEW

**Comment by JayjeetAtGithub** [comment]
@ivotron This pr is not yet ready for reviewing.

**Comment by JayjeetAtGithub** [comment]
@ivotron This PR is ready for initial review. Please suggest me changes that i might have missed. Thanks !

File: ci/test/singularity [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  uses = "./actions/example"
+  args = ["ls -a"]
+  secrets = ["SECRET_KEY"]
+  env = {
+    F_NAME = "first_name",
+    L_NAME = "last_name"
+  }
+  needs = [
+    "busybox"
+  ]
```
[the remark this answers is not in the document]
Comment (JayjeetAtGithub) [reply ci/test/singularity]: @ivotron By public repository do you mean something like `actions/bin/sh@master` that we used for `DockerRunner` ?

[the remark this answers is not in the document]
Comment (ivotron) [reply ci/test/singularity]: yes, but for a singularity action. So we might need to create a repo containing such an action.

[the remark this answers is not in the document]
Comment (JayjeetAtGithub) [reply ci/test/singularity]: Thanks a lot !

**Comment by JayjeetAtGithub** [comment]
@ivotron Updated the tests ! Please have a look

**Comment by JayjeetAtGithub** [comment]
@ivotron The tests have actually passed , but its showing pending to me ? Please open [this](https://travis-ci.org/systemslab/popper/builds/511539131?utm_source=github_status&utm_medium=notification) to see the logs. Cant understand why this is happening ?

**Comment by ivotron** [comment]
thanks a lot!

**Pull request merged**